### DISCUSSION TITLE
Record pagination on page

### DISCUSSION THREAD

**Discussion by aeq-dev** [filamentphp/filament#3981] [Help] [answered]
Hello,
Sorry for this stupid question but I want to get record pagination for custom page, 
I did something like this : 
```

    public function mount(Module $record)
    {
        $this->lessons = $record->lessons()->paginate(10);
        ......
```
And the error says : 
`Livewire component's [app.filament.pages.study-module] public property [lessons] must be of type: [numeric, string, array, null, or boolean]. Only protected or private properties can be set as other types because JavaScript doesn't need to access them.`
I know this is a livewire issue as I searched, so how can I get this data on page render function ?
Thanks

**Comment by danharrin** [comment, marked as answer]
You can use

```php
protected function getViewData(): array
{
    return [
        'lessons' => $this->record->lessons()->paginate(10),
    ];
}
```

**Comment by aeq-dev** [comment]
Thank you dear @danharrin :)
